pkg/api/v1/defaults.go

```diff
@@ -160,6 +160,18 @@ func SetDefaults_Pod(obj *Pod) {
 			}
 		}
 	}
+	for i := range obj.Spec.InitContainers {
+		if obj.Spec.InitContainers[i].Resources.Limits != nil {
```

Comment:
Just not sure which "case/proposal" ask to do that :). In other word, what may happen if we do it in `pod.spec`?

not debt, just want to learn more history & cases :).